kroxylicious-operator/src/test/java/io/kroxylicious/operator/ProxyReconcilerTest.java

```diff
@@ -0,0 +1,112 @@
+import static org.awaitility.Awaitility.await;
+
+class ProxyReconcilerTest {
+
+    public static final String RESOURCE_NAME = "test-proxy";
+    public static final String INITIAL_BOOTSTRAP = "my-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092";
+    public static final String CHANGED_BOOTSTRAP = "your-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092";
+
+    @RegisterExtension
+    static LocallyRunOperatorExtension extension = LocallyRunOperatorExtension.builder()
+            .withReconciler(ProxyReconciler.class)
+            .build();
+
+    @Test
+    void testCRUDOperations() {
+        final var cr = extension.create(testResource());
+
+        await().untilAsserted(() -> {
+            var secret = extension.get(Secret.class, ProxyConfigSecret.secretName(cr));
+            assertThat(secret)
+                    .isNotNull()
+                    .extracting(ProxyReconcilerTest::decodeSecretData, InstanceOfAssertFactories.map(String.class, String.class))
+                    .containsKey(ProxyConfigSecret.configYamlKey(cr))
+                    .extracting(map -> map.get(ProxyConfigSecret.configYamlKey(cr)), InstanceOfAssertFactories.STRING)
+                    .contains(INITIAL_BOOTSTRAP)
+                    .doesNotContain(CHANGED_BOOTSTRAP);
+
+            var deployment = extension.get(Deployment.class, ProxyDeployment.deploymentName(cr));
+            assertThat(deployment).isNotNull()
+                    .extracting(dep -> dep.getSpec().getTemplate().getSpec().getVolumes(), InstanceOfAssertFactories.list(Volume.class))
+                    .describedAs("Deployment template should mount the proxy config secret")
+                    .anyMatch(vol -> vol.getSecret() != null
+                            && vol.getSecret().getSecretName().equals(ProxyConfigSecret.secretName(cr)));
+
+            var service = extension.get(Service.class, ProxyService.serviceName(cr));
+            assertThat(service).isNotNull()
+                    .extracting(svc -> svc.getSpec().getSelector())
+                    .describedAs("Service's selector should select proxy pods")
+                    .isEqualTo(ProxyDeployment.podLabels());
+        });
+
+        cr.getSpec().setBootstrapServers(CHANGED_BOOTSTRAP);
+        final var changedCr = extension.replace(cr);
+
+        await().untilAsserted(() -> {
+            var secret = extension.get(Secret.class, ProxyConfigSecret.secretName(cr));
+            assertThat(secret)
+                    .isNotNull()
+                    .extracting(ProxyReconcilerTest::decodeSecretData, InstanceOfAssertFactories.map(String.class, String.class))
+                    .containsKey(ProxyConfigSecret.configYamlKey(changedCr))
+                    .extracting(map -> map.get(ProxyConfigSecret.configYamlKey(changedCr)), InstanceOfAssertFactories.STRING)
+                    .doesNotContain(INITIAL_BOOTSTRAP)
+                    .contains(CHANGED_BOOTSTRAP);
```

Comment:
What is actually making the change? I don't see any code in the reconciler that actually makes this happen...

Reply:
1. A reconciliation happens via the watch because the `KafkaProxy` resource changed. 
2. The framework calls each of the dependent resources declared using `@Dependent` on the `ProxyReconciler`.
3. `ProxyConfigSecret#desired()` returns the new `Secret`
4. `ProxyReconciler#reconcile()` is called by the framework, which returns KafkaProxy with which the framework should match the CR in Kube.

I renamed the test to `ProxyReconcilerIT` to avoid readers having the expectation of being able to see this logic in the `ProxyReconciler`.